pkg/antctl/raw/supportbundle/command.go

```diff
@@ -523,8 +580,41 @@ func controllerRemoteRunE(cmd *cobra.Command, args []string) error {
 		defer f.Close()
 		io.Copy(f, reader)
 	}
-	if err := requestAll(agentClients, controllerClient, bar); err != nil {
+	results := requestAll(agentClients, controllerClient, bar)
+	resultMap := downloadAll(agentClients, controllerClient, dir, bar, results)
+
+	return processResults(resultMap, dir)
+}
+
+// processResults will output the failed nodes and their reasons if any. If no data was collected,
+// error is returned, otherwise will return nil.
+func processResults(resultMap map[string]error, dir string) error {
+	resultStr := ""
+	var failedNodes []string
+	allFailed := true
+	var err error
+
+	for k, v := range resultMap {
+		if k != "*" && v != nil {
+			resultStr += fmt.Sprintf("- %s: %s\n", k, v.Error())
+			failedNodes = append(failedNodes, k)
+		}
+		if v == nil {
+			allFailed = false
+		}
+	}
+	if resultStr != "" {
+		fmt.Println("Failed nodes: ")
+		fmt.Print(resultStr)
+	}
+
+	if failedNodes != nil {
+		err = writeFailedNodes(dir, failedNodes)
+	}
+
+	if allFailed {
+		return fmt.Errorf("no data was collected: %s", resultMap["*"].Error())
```

Comment:
It seems that we will mark allFailed true when we fail to collect either controller or agent logs. I don't think it is appropriate to record controller error here as it can be nil, right?

Reply:
No. allFailed = true means all node failed an controller failed.

Reply:
Right, allFailed means both controller and agents failed. Then what about the case  `antctl supportbundle node_a node_b node_c`, there is no controller bundle needed in this case. Will it cause a null pointer here if all the agents fail in this case?

Reply:
You are right!. I didn't run all the cases in every commit before. A new commit is added to address this issue